libs/community/langchain_community/document_loaders/mintbase.py

```diff
@@ -0,0 +1,188 @@
+import os
+import re
+import time
+from enum import Enum
+from typing import List, Optional
+
+import requests
+import json
+from langchain_core.documents import Document
+
+from langchain_community.document_loaders.base import BaseLoader
+
+
+class BlockchainType(Enum):
+    """Enumerator of the supported blockchains."""
+
+    NEAR_MAINNET = "mainnet"
+    NEAR_TESTNET = "testnet"
+
+
+class MintbaseDocumentLoader(BaseLoader):
+    """Load elements from a blockchain smart contract.
+
+    The supported blockchains are: Near mainnet, Near testnet.
+
+    If no BlockchainType is specified, the default is Near mainnet.
+
+    The Loader uses the Mintbase API to interact with the blockchain.
+    MB_API_KEY environment variable must be set to use this loader.
+
+    The API returns 100 NFTs per request and can be paginated using the
+    startToken parameter.
+
+    If get_all_tokens is set to True, the loader will get all tokens
+    on the contract.  Note that for contracts with a large number of tokens,
+    this may take a long time (e.g. 10k tokens is 100 requests).
+    Default value is false for this reason.
+
+    The max_execution_time (sec) can be set to limit the execution time
+    of the loader.
+
+    Future versions of this loader can:
+        - Support additional Mintbase APIs (e.g. getTransactions, etc.)
+    """
+
+    def __init__(
+        self,
+        contract_address: str,
+        blockchainType: BlockchainType = BlockchainType.NEAR_MAINNET,
```

Comment:
1) Favor Literal over Enums so user's don't have to import extra enums
2) snake_case for python

```suggestion
        blockchain_type: Literal['mainnet', 'testnet']
```